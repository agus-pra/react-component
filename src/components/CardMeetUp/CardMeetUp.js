import React from 'react';

import Avatar from '../Avatar/Avatar';
// import Button from '../Button/Button';
import AvatarImage from '../../assets/avatar.jpg';
import {Button, Wrapper, Table, Tr, Td, Card, Img} from './cardMeetUp.style.js';

const CardMeetUp = ({location, members, organizer}) => {

  const thanksForJoin = () => {
    alert('Terima kasih sudah join meetup');
  }

  return (
      <Wrapper>
        <Img>
          <Avatar src={AvatarImage} alt="img avatar" />
        </Img>
        <Card>
          <h2>Hacktiv 8 Meetup</h2>
          <Table>
            <Tr>
              <Td>Location</Td><Td>{location}</Td>
            </Tr>
            <Tr>
              <Td>Members</Td><Td>{members}</Td>
            </Tr>
            <Tr>
              <Td>Organizer</Td><Td>{organizer}</Td>
            </Tr>
          </Table>
          <Button onClick={thanksForJoin}>JOINT US</Button>
          <Button primary onClick={thanksForJoin}>START</Button>
        </Card>
      </Wrapper>
  );
}

export default CardMeetUp;