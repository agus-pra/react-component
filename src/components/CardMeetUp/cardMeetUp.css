.cardJoinUs {
  display: flex;
  justify-content: flex-start;
  background-color: #8E8C8C;
  color: #fff;
  margin: 10px 10px 10px 10px;
  padding: 10px;
}

.cardJoinUs2 {
  padding-left: 30px;
  text-align: left;
}

.cardJoinUs2 table {
  text-align: left;
}