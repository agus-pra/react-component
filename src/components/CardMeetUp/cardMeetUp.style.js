import styled from 'styled-components';

export const Wrapper = styled.section`
display: flex;
justify-content: flex-start;
background-image: linear-gradient(to right, #00F260, #0575E6);
box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
color: #fff;
margin:10px 10px 10px 10px;
padding: 10px;
`;

export const Img = styled.section`
padding-top: 20px;
`;

export const Table = styled.table`
  border-collapse: collapse; 
  width: 100%; 
`;

export const Tr = styled.tr`
    text-align: left;
    padding: 8px;
`;

export const Td = styled.td`
    text-align: left;
    padding: 8px;
`;

export const Card = styled.section`
text-align: left;
padding: 8px;
`;

export const Button = styled.button`

/* Adapt the colors based on primary prop */

background: ${props => props.primary ? "palevioletred" : "white"};
color: ${props => props.primary ? "white" : "palevioletred"};

font-size: 1em;
margin: 1em;
padding: 0.25em 1em;
border: 2px solid palevioletred;
border-radius: 3px;
`;