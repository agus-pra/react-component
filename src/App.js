import React from 'react';

import CardMeetUp from './components/CardMeetUp/CardMeetUp';
import MeetUp from './components/NextMeetUp/MeetUp';
import Members from './components/Members/Members';
import PastMeet from './components/PastMeetUp/PastMeet';
import './App.css';

function App() {
  const dataPastMeet = [
    {
      date: '27 Nov 2017',
      deskripsi: '#39 JakartaJS April Meetup with Kumparan',
      went:'137',
      btn :'primary',
    },
    {
      date: '27 Oct 2018',
      deskripsi: '#38 JakartaJS April Meetup with Blibli',
      went:'137',
      btn :'success',
    },
    {
      date: '27 Jam 2019',
      deskripsi: '#37 JakartaJS April Meetup with Hacktiv8',
      went:'137',
      btn :'warning',
    },
  ]

  return (
    <div className="App">
      <div className="menu">
        <ul>
          <li><a class="active" href="#home">Home</a></li>
          <li><a href="#news">News</a></li>
          <li><a href="#contact">Contact</a></li>
          <li><a href="#about">About</a></li>
        </ul>
      </div>

      {/* Call component */}
      <CardMeetUp location="Medan, Indonesia" members="180" organizer="ReactJs"/>
      <h2>Next MeetUp</h2>

      <MeetUp title="Awesome Meetup and event"/>
      <h2>About Meetup</h2>
        <div className="text1">
          <p> 
            Come and meet other developers interested in the JavaScript and its library in the Greater Jakarta area<br /><br />
            Twitter: @JakartaJS and we user the hashtag #jakartajs
          </p>
          <br />
        </div>
        <div>
          <h2>Members</h2> 
          <text className="SeeAll">See all</text><br />
        </div>

    < Members name="Agus Prabowo" sum="105 .others" />
    <h2>Past Meetup</h2> 
    <text className="SeeAll">See all</text><br />

    <div className="cartPastMeet">
    <PastMeet data={dataPastMeet}/>  
    </div>

  </div>
  );
}

export default App;
