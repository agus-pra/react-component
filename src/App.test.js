import React from 'react';

import { shallow, mount } from 'enzyme';
import toJson from "enzyme-to-json";
import CardMeetUp from './components/CardMeetUp/CardMeetUp';

import App from './App';

it('Pages App running without crash', () => {
  shallow(<App />);
});

describe('Test Header CardMeetup', () => {
  it('accept user account props', () => {
    const wrapper = mount(<CardMeetUp location="Medan, Indonesia" members="180" organizer="ReactJs" />);
    const heading = wrapper.find("h2").text();
    expect(heading).toEqual("Hacktiv8 Meetup");
  });
});


// describe('Test component CardMeetup', () => {
//   it('accept component', () => {
//     const wrapper = mount(<CardMeetUp location="Medan, Indonesia" members="180" organizer="ReactJs" />);
//     const   Td = wrapper.find("Td").text();
//     expect(Td).toEqual("Location");
//   });
// });


it("renders correctly", () => {
  const tree = mount(<CardMeetUp location="Medan, Indonesia" members="180" organizer="ReactJs" />);
  expect(toJson(tree)).toMatchSnapshot();
});
