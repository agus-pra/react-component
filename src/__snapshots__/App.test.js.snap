// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<CardMeetUp
  location="Medan, Indonesia"
  members="180"
  organizer="ReactJs"
>
  <styled.section>
    <section
      className="sc-bdnylx cekMnc"
    >
      <styled.section>
        <section
          className="sc-gtssRu XWBRl"
        >
          <Avatar
            alt="img avatar"
            src="avatar.jpg"
          >
            <img
              alt="img avatar"
              className="avatar"
              src="avatar.jpg"
            />
          </Avatar>
        </section>
      </styled.section>
      <styled.section>
        <section
          className="sc-jSFkmK iWYDKa"
        >
          <h2>
            Hacktiv8 Meetup
          </h2>
          <styled.table>
            <table
              className="sc-dlnjPT edvPjk"
            >
              <styled.tr>
                <tr
                  className="sc-hKFyIo brbWtm"
                >
                  <styled.td>
                    <td
                      className="sc-eCApGN jdectq"
                    >
                      Location
                    </td>
                  </styled.td>
                  <styled.td>
                    <td
                      className="sc-eCApGN jdectq"
                    >
                      Medan, Indonesia
                    </td>
                  </styled.td>
                </tr>
              </styled.tr>
              <styled.tr>
                <tr
                  className="sc-hKFyIo brbWtm"
                >
                  <styled.td>
                    <td
                      className="sc-eCApGN jdectq"
                    >
                      Members
                    </td>
                  </styled.td>
                  <styled.td>
                    <td
                      className="sc-eCApGN jdectq"
                    >
                      180
                    </td>
                  </styled.td>
                </tr>
              </styled.tr>
              <styled.tr>
                <tr
                  className="sc-hKFyIo brbWtm"
                >
                  <styled.td>
                    <td
                      className="sc-eCApGN jdectq"
                    >
                      Organizer
                    </td>
                  </styled.td>
                  <styled.td>
                    <td
                      className="sc-eCApGN jdectq"
                    >
                      ReactJs
                    </td>
                  </styled.td>
                </tr>
              </styled.tr>
            </table>
          </styled.table>
          <styled.button
            onClick={[Function]}
          >
            <button
              className="sc-gKAblj hLYzZy"
              onClick={[Function]}
            >
              JOINT US
            </button>
          </styled.button>
          <styled.button
            onClick={[Function]}
            primary={true}
          >
            <button
              className="sc-gKAblj ebwLEC"
              onClick={[Function]}
            >
              START
            </button>
          </styled.button>
        </section>
      </styled.section>
    </section>
  </styled.section>
</CardMeetUp>
`;
